### PR TITLE
Colorblind syntax colors

### PR DESCRIPTION
**Pull request by NikolayXHD**
**Attention this PR is made above #8424 so only last 2 commits are to be reviewed here**

Correct color calculations to make result colorblind-friendly If "Colorblind" checkbox is set in ColorSettings page, replace red-ish colors with blue-ish by compressing blue-orange ark in hue circle into blue-violet

image

before
image

after
image

after v2, modified to reduce contrast loss, as @neil5280 noticed

image

Besides syntax highlighting colors, other auto-calculated colors get affected as well.
Observe how build status colors change

before
image

after
image

after v2
image

### CODE REVIEW

**Review by gerhardol — COMMENTED** [review]
The changes look fine to me.
The looks is not great, but this change is not for me anyway. (I always prefer usability before looks.)

@itsho @neil5280 
Is this something you see as an improvement?

**Comment by NikolayXHD** [comment]
> The changes look fine to me.
> The looks is not great, but this change is not for me anyway. (I always prefer usability before looks.)

From what you say I guess I did not describe the PR change clearly. The colors are only altered if "Colorblind" checkbox is set.

**Comment by gerhardol** [comment]
> From what you say I suspect I did not describe the PR change clearly. The colors are only altered if "Colorblind" checkbox is set.

The description is clear. The only I can say about the colors (normal color view for me) is that they do not look good - but this is not for me.
The reasoning about the change makes sense to me.

**Comment by neil5280** [comment]
> The changes look fine to me.
> The looks is not great, but this change is not for me anyway. (I always prefer usability before looks.)
> 
> @itsho @neil5280
> Is this something you see as an improvement?

For my color blindness, the shift in colors for the difference viewer were not advantageous, making the changed code look like unchanged code:

image

The shifts in the color on the build status were helpful.

This highlights the difficulties in creating a one-size-fits-all automatic adjustment for those affected by color-blindness.

**Comment by NikolayXHD** [comment]
> For my color blindness, the shift in colors for the difference viewer were not advantageous, making the changed code look like unchanged code:
> 
> image

Adjusted the formula to reduce color contrast loss,
image

**Review by hani77979 — APPROVED** [review]

**Comment by hani77979** [comment]
G

**Comment by RussKie** [comment]
I plan to review this in the next few days.

**Comment by NikolayXHD** [comment]
> I plan to review this in the next few days.

@RussKie then I hold this PR and also #8424

**Comment by RussKie** [comment]
@NikolayXHD thank you for doing this!
If this doesn't affect the default theming, and only affects users with visual impairments - I defer approval to them.

**Comment by NikolayXHD** [comment]
> @NikolayXHD thank you for doing this!
> If this doesn't affect the default theming, and only affects users with visual impairments - I defer approval to them.

@neil5280, @itsho It's up to your approve now. See screenshots in PR title, they are up-to-date with current PR code.

**Comment by itsho** [comment]
I've asked my colorblind friend and he says it looks great!
Thank you - to everyone involved in this!

**Comment by neil5280** [comment]
Contrast is improved.  Thanks for asking.

**Review by RussKie — CHANGES_REQUESTED** [review]
Blocking until #8424 is merged.

**Pull request merged**
